### PR TITLE
camera shake bugfix

### PR DESCRIPTION
**Pull request by Mpcs**
# Description
Fix for the bug causing a camera shake when changing the ship with one with smaller max HP.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Mpcs** [comment]
Turns out there was a bug when the ship was transcended. Fixed.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/destinationsol/game/SolCam.java [hunk 1]
```diff
@@ -47,6 +48,7 @@
     private final Vector3 myTmpVec;
 
     private float myPrevHeroLife;
+    private Hull myPrevHeroHull;
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/game/SolCam.java]:
```suggestion
    private Hull previousHeroHull;
```
We are trying to get rid of all these `myX` variable names as well as the abbreviations.

File: engine/src/main/java/org/destinationsol/game/SolCam.java [hunk 2]
```diff
@@ -74,8 +76,13 @@ public void update(SolGame game, float timeStep) {
             return;
         }
 
+
         Hero hero = game.getHero();
         float life = hero.getLife();
+        Hull hull = null;
+        if(hero.isNonTranscendent())
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/game/SolCam.java]:
```suggestion
        if (hero.isNonTranscendent())
```
Also please use curly braces even if the block is a single line

File: engine/src/main/java/org/destinationsol/game/SolCam.java [hunk 3]
```diff
@@ -94,14 +101,14 @@ public void update(SolGame game, float timeStep) {
             }
         }
 
-        if (life < myPrevHeroLife) {
+        if (life < myPrevHeroLife && hull == myPrevHeroHull) {
```
Comment (NicholasBatesNZ) [inline engine/src/main/java/org/destinationsol/game/SolCam.java]: Looking just at the code, it is not immediately obvious why this comparison fixes the bug. Could you please add a comment here

File: engine/src/main/java/org/destinationsol/game/SolCam.java (hunk 1 above)
Comment (Mpcs) [reply engine/src/main/java/org/destinationsol/game/SolCam.java]: Okay, I thought I'll stay consistent with the naming

**Comment by Mpcs** [comment]
Should be okay now.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request merged**